packages/models-library/src/models_library/services_resources.py

```diff
@@ -1,10 +1,28 @@
-from typing import Union
+import logging
+from typing import Any, Union
 
 from models_library.generics import DictModel
-from pydantic import BaseModel, StrictFloat, StrictInt, root_validator
+from pydantic import (
+    BaseModel,
+    ByteSize,
+    Field,
+    StrictFloat,
+    StrictInt,
+    constr,
+    root_validator,
+)
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_RESOURCES: dict[str, Any] = {
+    "CPU": {"limit": 0.1, "reservation": 0.1},
+    "RAM": {"limit": ByteSize(2 * 1024**3), "reservation": ByteSize(2 * 1024**3)},
+}
 
 ResourceName = str
 
+ComposeImage = constr(regex=r"[\w/-]+:[\w.@]+")
```

Comment:
so it is actually we use ServiceKey and ServiceVersion. Look for them in the code.  key:version

Reply:
Because this can be either a simcore service or just some other dockerhub available image. It's also the `image` key inside the docker-compose service.